dataplane/standalone/apigen/apigen.go

```diff
@@ -260,6 +261,248 @@ var (
+	ResponseFields             []protoTmplField
+}
+
+type protoTmplField struct {
+	ProtoType string
+	Name      string
+	Index     int
+}
+
+type protoRPC struct {
+	RequestName  string
+	ResponseName string
+	Name         string
+}
+
+type saiTypeInfo struct {
+	Repeated  bool
+	Primitive bool
+	ProtoType string
+}
+
+var saiTypeToProto = map[string]saiTypeInfo{
+	"bool": {
+		ProtoType: "bool",
+	},
+	"char": {
+		ProtoType: "bytes",
+	},
+	"sai_uint8_t": {
+		ProtoType: "uint32",
+	},
+	"sai_int8_t": {
+		ProtoType: "int32",
+	},
+	"sai_uint16_t": {
+		ProtoType: "uint32",
+	},
+	"sai_int16_t": {
+		ProtoType: "int32",
+	},
+	"sai_uint32_t": {
+		ProtoType: "uint32",
+	},
+	"sai_int32_t": {
+		ProtoType: "uint32",
+	},
+	"sai_uint64_t": {
+		ProtoType: "uint64",
+	},
+	"sai_int64_t": {
+		ProtoType: "int64",
+	},
+	"sai_mac_t": {
+		ProtoType: "bytes",
+	},
+	"sai_ip4_t": {
+		ProtoType: "bytes",
+	},
+	"sai_ip6_t": {
+		ProtoType: "bytes",
+	},
+	"sai_s32_list_t": {
+		Repeated:  true,
+		ProtoType: "int32",
+	},
+	"sai_object_id_t": {
+		ProtoType: "uint64",
+	},
+	"sai_object_list_t": {
+		Repeated:  true,
+		ProtoType: "uint64",
+	},
+	"sai_encrypt_key_t": {
+		ProtoType: "bytes",
+	},
+	"sai_auth_key_t": {
+		ProtoType: "bytes",
+	},
+	"sai_macsec_sak_t": {
+		ProtoType: "bytes",
+	},
+	"sai_macsec_auth_key_t": {
+		ProtoType: "bytes",
+	},
+	"sai_macsec_salt_t": {
+		ProtoType: "bytes",
+	},
+	"sai_u32_list_t": {
+		Repeated:  true,
+		ProtoType: "uint32",
+	},
+	"sai_segment_list_t": {
+		Repeated:  true,
+		ProtoType: "bytes",
+	},
+	"sai_s8_list_t": {
+		Repeated:  true,
+		ProtoType: "int32",
+	},
+	"sai_u8_list_t": {
+		Repeated:  true,
+		ProtoType: "uint32",
+	},
+	"sai_port_err_status_list_t": {
+		Repeated:  true,
+		ProtoType: "sai_port_err_status_t",
+	},
+	"sai_vlan_list_t": {
+		Repeated:  true,
+		ProtoType: "uint32",
+	},
+	"sai_u32_range_t": {
+		ProtoType: "Uint32Range",
+	},
+	"sai_ip_address_t": {
+		ProtoType: "IpAddress",
+	},
+	"sai_map_list_t": {
+		Repeated:  true,
+		ProtoType: "Uint32Range",
+	},
+	"sai_tlv_list_t": {
+		Repeated:  true,
+		ProtoType: "TLV",
+	},
+	"sai_qos_map_list_t": {
+		Repeated:  true,
+		ProtoType: "QOSMap",
+	},
+	"sai_system_port_config_t": {
+		Repeated:  true,
+		ProtoType: "SystemPortConfig",
+	},
+	"sai_system_port_config_list_t": {
+		Repeated:  true,
+		ProtoType: "SystemPortConfig",
+	},
+	"sai_ip_address_list_t": {
+		Repeated:  true,
+		ProtoType: "IpAddress",
+	},
+	"sai_port_eye_values_list_t": {
+		Repeated:  true,
+		ProtoType: "PortEyeValues",
+	},
+	"sai_prbs_rx_state_t": {
+		ProtoType: "PRBS_RXState",
+	},
+	"sai_fabric_port_reachability_t": {
+		ProtoType: "FabricPortReachability",
+	},
+	"sai_acl_resource_list_t": {
+		Repeated:  true,
+		ProtoType: "ACLResource",
+	},
+	"sai_acl_capability_t": {
+		Repeated:  true,
+		ProtoType: "ACLCapability",
+	},
+}
+
+var saiTypeToProtoTypeCompound = map[string]func(next string, xmlInfo *xmlInfo) string{
```

Comment:
I know this is prototype code, but should we add some comments here to explain what `next` is and what this function is doing?

Reply:
done